src/main/java/nz/ac/auckland/scriptella/driver/http/HTTPConnection.java

```diff
@@ -1,288 +0,0 @@
-                } else {
-                    httpRequestBase = new HttpPost(host);
-                    logger.debug("Http method is POST.");
-                }
-
-                if (format.toUpperCase().equals("STRING")) {
-
-                    httpRequestBase.setEntity(new UrlEncodedFormEntity(generateParams(resource, parameters)));
-                    logger.debug("URLEncodedFormEntity created and set.");
-                } else { // JSON
-
-                    StringEntity se = new StringEntity(parseJSON(resource, parameters), "UTF-8");
-                    logger.debug("JSON parsed.");
-
-                    se.setContentType("application/json; charset=UTF-8");
-
-                    httpRequestBase.setEntity(se);
-                    logger.debug("JSON format entity set for http request.");
-                }
-
-                httpResponse = httpClient.execute(httpRequestBase);
-                logger.debug("Http request executed.");
-            }
-
-            logger.info("Response Status: {}", httpResponse.getStatusLine().getStatusCode());
-
-        } catch (IOException | URISyntaxException e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-
-    private List<NameValuePair> generateParams(Resource resource, ParametersCallbackMap parameters) {
-        BufferedReader br = null;
-
-        try {
-            br = new BufferedReader(resource.open());
-            logger.debug("Resource opened for reading.");
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
-
-        String line;
-        List<NameValuePair> nameValuePairList = new ArrayList<>(1);
-
-        try {
-            while ((line = br.readLine()) != null) {
-                String[] components = line.trim().split("=");
-                if (components.length > 1) {
-                    // Remove "$" and double-quotes used to make driver work as user would expect based on csv driver and others.
-
-                    String key = components[1].replace("$", "").replace("?", "").replace("\"", "");
-                    logger.debug("Stripped \'$\' and \'?\' from variables.");
-                    if (parameters.containsKey(key)) {
-                        nameValuePairList.add(new BasicNameValuePair(components[0], (String) parameters.getParameter(key)));
-
-                        logger.debug("Variable found, added to List.");
-                    } else {
-                        nameValuePairList.add(new BasicNameValuePair(components[0], components[1]));
-                        logger.debug("Line contains no variables, adding to list as literal.");
-                    }
-                }
-            }
-        } catch (IOException e) {
-            logger.error("Error occurred while generating request parameters: ", e);
-            throw new RuntimeException(e);
-        }
-
-        return nameValuePairList;
-    }
-
-
-    private String parseJSON(Resource resource, ParametersCallbackMap parameters) {
```

Comment:
This method does three things: read from resource, load resource into list, process list.
You should make it as two or three small methods, and write unit test for each one.